### ISSUE TITLE
Regression: Block crashes when inserting a link and going through the links suggestions with the arrow keys

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#24037]
**Describe the bug**

When inserting a link, navigating through the list of link suggestions with the Down / Up arrow keys makes the block crash.

I can't reproduce this behavior in WordPress 5.4 so this appears to be a regression in WordPress 5.5 Beta 2.

**To reproduce**
- test with WordPress 5.5 Beta 2
- select a word in a paragraphs text 
- press Cmd + K (Ctrl + K on Windows)
- the popup to insert a link appears
- type something in the search to get some results
- use the Down arrow key to navigate through the suggested links
- once on the last suggestion, press the Down arrow key one more
- the block crashes
- repeat the steps above but this time use the Up arrow key
- it should start navigating the suggestions from the bottom
- the block crashes after the first Down arrow key press

The error shown in the console is:
`Uncaught TypeError: Cannot read property 'ownerDocument' of null`

Animated GIF:

![insert link block crash](https://user-images.githubusercontent.com/1682452/87878668-09a9b780-c9e6-11ea-8426-b2705a79bfeb.gif)

**Comment by afercia** [comment]
Thanks for the quick fix 👍 

However, I still see something that doesn't look right to me.

- search for a link
- in my example, I get two visible results
- I use the Down arrow key to highlight the last result
- when I press the Down arrow key again, it doesn't "loop" any longer: it should circle back and start highlighting the results from the first one
- instead, it seems the third Down arrow key press doesn't do anything
- pressing the Down arrow key a fourth time, does loop
- in WordPress 5.4 the "looping" works correctly

Wild guess: it seems there's something wrong in some length calculation, as when using a screen reader it announces "3 results found" but the visible results are only 2. Screenshot:

<img width="939" alt="Screenshot 2020-07-20 at 14 45 16" src="https://user-images.githubusercontent.com/1682452/87939870-fe639400-ca98-11ea-8e67-9363829171bc.png">

Reopening. Please do let me know if you want me to open a new issue instead.

**Comment by afercia** [comment]
Adding the accessibility label as this issue has a greater impact for keyboard users and screen reader users.

**Comment by youknowriad** [comment]
I'm not able to reproduce, for me it loops properly.

**Comment by afercia** [comment]
I just tested again, on two different machines, and I'm still able to reproduce. Reopening.

Screenshot from a Windows machine:

![Screenshot (120)](https://user-images.githubusercontent.com/1682452/88553413-8ca1c200-d025-11ea-896f-137bf6cc0970.png)

- there are 3 visible search results
- the spoken message says there are 4 results
- there's an "empty" arrow key press that doesn't do anything

Again, it seems there's some wrong count, which might also explain the "emoty" arrow key press.
